app/src/main/java/org/dslul/openboard/inputmethod/latin/settings/AdvancedSettingsFragment.kt

```diff
@@ -89,13 +89,18 @@ class AdvancedSettingsFragment : SubScreenFragment() {
         if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.R) {
             removePreference(Settings.PREF_SHOW_SETUP_WIZARD_ICON)
         }
+        // Write this here once to avoid a ghost effect about the Debug settings menu
+        if (!BuildConfig.DEBUG && !sharedPreferences.getBoolean(DebugSettings.PREF_SHOW_DEBUG_SETTINGS, false)) {
+            removePreference(Settings.SCREEN_DEBUG)
+        }
```

Comment:
Is this actually necessary? `onStart` should be called shortly after `onCreate`, still before you see the UI.

Reply:
After my tests, unfortunately  yes, to avoid this ghost effect.

Reply:
I tried without this (i.e. only adding the `onStart` after `onCreate(null)`) and it works for me.
The necessity to remove the setting both in `onStart` and `onCreate` doesn't seem logical to me...

Reply:
I'm finally taking some time to test this.
Can you confirm that you don't see this ghost effect only when calling onStart() after onCreate()??
On both my devices, the menu appears for 1 or 2 seconds and then disappears.
I should point out that this happens **without having performed a restore**.

Reply:
If I do not restore anything, I don't see the debug settings.

Reply:
We can therefore conclude that this is specific to certain devices. Very strange...
For the moment, as discussed I'm just adding the `onStart()` value after `onCreate()` --> commit 51c2fe6